File: csharp/darts/Darts.cs
using System;

public static class Darts
{
    public static int Score(double x, double y)
    {
        var radius = Math.Sqrt(x * x + y * y);

        if (radius <= 1.0)
            return 10;
        if (radius <= 5.0)
            return 5;
        if (radius <= 10.0)
            return 1;
        return 0;
    }
}

File: kotlin/darts/src/main/kotlin/Darts.kt
import kotlin.math.ceil
import kotlin.math.sqrt

object Darts {

    fun score(x: Number, y: Number): Int = when (distance(x.toDouble(), y.toDouble())) {
        in 0..1 -> 10
        in 2..5 -> 5
        in 6..10 -> 1
        else -> 0
    }

    private fun distance(x: Double, y: Double): Int {
        val r = sqrt(x * x + y * y)
        return ceil(r).toInt()
    }
}
